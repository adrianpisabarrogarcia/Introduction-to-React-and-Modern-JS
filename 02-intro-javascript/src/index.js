//Desestructuración
//Asignación Desestructurante


const person = {
    name: 'Carlos',
    age: 30,
    key: 'Ironman'
}

//En react la mayoría de veces utilizamos los const
const {name:name2} = person;
const {name, age, key} = person;


console.log(person.name)
console.log(person.age)
console.log(person.key)


console.log(name2)
//is deprecated: use destructuring assignment instead
console.log(name)

