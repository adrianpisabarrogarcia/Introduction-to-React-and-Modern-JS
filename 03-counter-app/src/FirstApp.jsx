//rafc <- atajo para crear componentes

//import { Fragment } from "react"

const newMessage = 'Adrián' //<- esto es una variable que se usa en el componente
const newBoolean = true 
const newArray = [1,2,3,4,5,6,7,8,9]
const newObject = {
    message: "Hola Mundo",
    title: "Adrián P. G.",
}

const sayHello = (name) => `Hello ${name}!`


export const FirstApp = () => {
    //cuando tenga que volver a ejecutar para mostrar esta función,
    //vovlera a renderizar esta variable por eso no es recomedable usarla dentro de una función
    //sino que se debe usar en el componente (arriba ☝️)
    //const newMessage = 'Adrián' <- esto es una variable local que no se puede usar fuera de la función

    return(
        //<Fragment>
        <>
            <h2>First App</h2>
            <p>This is a paragraph 1</p>
            <p>This is a paragraph 2</p>
            <p>This is a paragraph 3</p>
            <p>This is a paragraph 4</p>
            <p>This is a paragraph 5</p>
            <p>This is a paragraph 6</p>
            <h1>{ 2 * 4 }</h1>
            <h1>{ newMessage }</h1>
            <h1>{ newBoolean }</h1>
            <h1>{ newArray }</h1>
            <code>{ JSON.stringify( newObject ) }</code>
            <h1>{ newObject.message }</h1>
            <h1>{ newObject.title }</h1>
            {/*
            comentario
            multilinea
            <h1>{ newObject.message }</h1>
            */}
            <h1 data-testid="test-title">{ sayHello('Adrián') }</h1>
        </>
        //</Fragment>
    )
}