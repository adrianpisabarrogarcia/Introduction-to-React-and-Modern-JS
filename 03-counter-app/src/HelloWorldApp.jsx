//option 1
/*
function App() {
    return (
        <div>
            <h1>Hello World!!!</h1>
        </div>
    )
}

export default App;
*/

// option 2
export const App = () => {
    return (
        <div>
            <h1>Hello World!!!</h1>
        </div>
    )
}

//option 3
//export const App = () => <h1>Hola Mundo!</h1>
