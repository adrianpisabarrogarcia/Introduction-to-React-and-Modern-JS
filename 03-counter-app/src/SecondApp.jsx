
export const SecondApp = ({ title = 'Hola Pepe!', subtitle, passNumber }) => {

    return (
        <>  
            <h3>{ title }</h3>
            <h3>{ subtitle }</h3>
            <h3>{ subtitle }</h3>
            <h3>{ passNumber }</h3>
        </>
    )
}
