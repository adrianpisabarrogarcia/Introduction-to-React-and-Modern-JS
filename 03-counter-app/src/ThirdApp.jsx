//añadir la dependencia en la terminal con yarn add prop-types, npm install prop-types
import PropTypes from "prop-types"

const ThirdApp = ({ title, subtitle }) => {
    
    //asi mejor no
    /*
    if (!title) {
        throw new Error('Title is required')
    }
    */

    return(
        <div>ThirdApp ➡️ { title} - { subtitle } </div>
    )
}

export default ThirdApp


ThirdApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}