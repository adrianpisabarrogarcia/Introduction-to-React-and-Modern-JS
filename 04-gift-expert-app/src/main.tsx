import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import { GiftExpertApp } from "./components/GiftExpertApp";

import "./styles.css";

createRoot(document.getElementById("root")!).render(
  <StrictMode>
    {/* titulo */}
    <GiftExpertApp />
    {/* Input */}

    {/* Listado de Gifs */}
  </StrictMode>
);
