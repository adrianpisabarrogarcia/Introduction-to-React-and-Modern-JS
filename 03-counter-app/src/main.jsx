import React from 'react'
import ReactDOM from 'react-dom/client'

//import App from './HelloWorldApp'
import { App } from './HelloWorldApp'
import { FirstApp } from './FirstApp'
import { SecondApp } from './SecondApp'
import ThirdApp from './ThirdApp'
import { CounterApp } from './CounterApp'
import './styles.css';

// Eventos de React: https://es.reactjs.org/docs/events.html

ReactDOM.createRoot(document.getElementById('root'))
    .render(
        <React.StrictMode>
            <App />
            <FirstApp />
            <SecondApp subtitle='Hola, Amaia!!' passNumber={1 + 3} />
            <ThirdApp subtitle="Hola, Amaia!!" />
            <hr></hr>
            <button onClick={(event) => console.log(event)}>
                Ver objeto evento
            </button>
            <button onClick={(event) => console.log(event)}>
                +1
            </button>
            <hr></hr>
            <CounterApp value={2 + 2} />

        </React.StrictMode>
    )

