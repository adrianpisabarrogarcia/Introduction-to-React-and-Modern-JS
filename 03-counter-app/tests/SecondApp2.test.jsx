import { render, screen } from "@testing-library/react";
import { SecondApp } from "../src/SecondApp"

describe('Pruebas en <SecondApp />', () => {

    const subTitle = 'Hola, Amaia!!'
    const passNumber = 4

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<SecondApp subtitle={subTitle} passNumber={passNumber} />)
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar el mensaje "Hola, Amaia!!"', () => {

        const { getAllByText } = render(<SecondApp subtitle={subTitle} passNumber={passNumber} />)
        //screen.debug()  // <- Muestra el HTML de la pantalla
        const subtitleElements = getAllByText(subTitle);
        expect(subtitleElements.length).toBe(2); // We expect exactly 2 elements with the subtitle
        expect(subtitleElements[0]).toBeTruthy();  // Check if at least one is found
        expect(subtitleElements[1]).toBeTruthy();  // Check the second occurrence
        expect(screen.getByText(passNumber)).toBeTruthy();
    })

    test('Debe de mostrar el titulo en un h3', () => {
        render(<SecondApp subtitle={subTitle} passNumber={passNumber} />);
        const headingElements = screen.getAllByRole('heading', { level: 3 });
        
        // Check if one of the h3 elements contains the subtitle
        const titleElement = headingElements.find((el) => el.innerHTML.includes(subTitle));
        expect(titleElement).toBeTruthy();
        expect(titleElement.innerHTML).toContain(subTitle);
    })

    test('Debe mosrtar el subtitulo enviado por props', () => { 
        render(<SecondApp subtitle={subTitle} passNumber={passNumber} />)
        expect(screen.getAllByText(subTitle).length).toBe(2);
    })
})