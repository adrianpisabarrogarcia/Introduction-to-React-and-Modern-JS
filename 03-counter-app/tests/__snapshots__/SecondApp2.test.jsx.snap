// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SecondApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h3>
    Hola Pepe!
  </h3>
  <h3>
    Hola, Amaia!!
  </h3>
  <h3>
    Hola, Amaia!!
  </h3>
  <h3>
    4
  </h3>
</div>
`;
